exporter/collector/logs.go

```diff
@@ -109,8 +115,39 @@ func (l *LogsExporter) Shutdown(ctx context.Context) error {
 }
 
 func (l *LogsExporter) PushLogs(ctx context.Context, ld plog.Logs) error {
+	batches, err := l.mapper.createBatches(ld)
+	if err != nil {
+		return err
+	}
+
+	// write batches
+	errors := []error{}
+	for _, batch := range batches {
+		_, err := l.writeLogEntries(ctx, batch)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if len(errors) > 0 {
+		return multierr.Combine(errors...)
+	}
+	return nil
+}
+
+func (l logMapper) createBatches(ld plog.Logs) ([][]*logpb.LogEntry, error) {
```

Comment:
nit: it might be cleaner to keep the logic simpler here by returning a `[]*logpb.LogEntry`, and then send it in chunks like the monitoring batching code does: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/metrics.go#L208

Reply:
Yeah that is how the Stanza implementation did it too

Reply:
Updated, we can't use the direct indexing that metrics does since this is based on size rather than count. But I think I was still able to refactor it to do this more in-place. Please let me know what you think

Reply:
ah, yeah.  That is unfortunate, but I think the current code is good